### PR TITLE
Simplification of CacheProxy.Read func.

### PR DESCRIPTION
**Pull request by vadimberezniker**
**Related issues**: N/A

### CODE REVIEW

File: enterprise/server/util/cacheproxy/cacheproxy.go [hunk 1]
```diff
@@ -328,17 +328,12 @@ func (c *CacheProxy) Read(req *dcpb.ReadRequest, stream dcpb.DistributedCache_Re
 		n, err := io.ReadFull(reader, copyBuf)
```
Comment (bduffany) [inline enterprise/server/util/cacheproxy/cacheproxy.go:328]: I feel like the error handling logic is still a bit confusing. It seems like the utility of `ReadFull` is that it simplifies error handling, but we are still explicitly inspecting the errors that it returns, so we are losing out on the benefit.

wdyt about something like

```
for remaining := rn.GetDigest.GetSizeBytes(); remaining > 0; {
  b := copyBuf[:min(len(copyBuf), remaining)]
  n, err := io.ReadFull(reader, b)
  if err != nil {
    return err // all errors are unexpected
  }
  // stream.Send({ Data: b })
  remainder -= n
}
```

Comment (vadimberezniker) [reply enterprise/server/util/cacheproxy/cacheproxy.go]: I don't think that's much better. I added a new ioutil func that works like io.ReadFull but does not return an error on a short read.

**Review by tylerwilliams — APPROVED** [review]

File: enterprise/server/util/cacheproxy/cacheproxy.go (hunk 1 above)
Comment (bduffany) [reply enterprise/server/util/cacheproxy/cacheproxy.go]: hm it seems like with this approach, we now need an additional Read() in order to discover the EOF? wondering if that would increase IO load

Comment (vadimberezniker) [reply enterprise/server/util/cacheproxy/cacheproxy.go]: There's no change. Previously ErrUnexpectedEOF did not terminate the loop, it was only terminated on the next EOF.

**Review by bduffany — APPROVED** [review]

**Pull request merged**